src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -37,14 +38,13 @@ SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients(
 {
 	SGVector<float64_t> vec = generate_random_parameter_vector();
 	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	random_params.zero();
```

Comment:
all this is overly complicated.
There should just be a boradcasting style method to fill a matrix with a given vector (either column or row) in linalg.
```
linalg::set(random_params, vec); // needs to be implemented
```

Or a single simple loop using memset, in the lines of (which would also be the implementation in linalg)
```
for i in range(num_samples)
  sg_memcpy(random_params.data() + i*nmum_samples, vec.data(), vec.clen);
```
The above could even be shortened with a c++ `for_each` or similar

Reply:
Okay there has been work on creating column iterator for SGMatrix. Once that's merged we can do 
``` std::for_each(random_params.begin_col(),random_params.end_col(),[](const auto& random_vec){ random_vec = generate_random_vector();} ```

Reply:
btw there is also a simple `set_column` method of SGMatrix

Reply:
Aye, this seems to do the trick. It's compiling well and unit tests pass.  Any changes you recommend before we finalize?